### PR TITLE
Jacdac updates

### PR DESCRIPTION
**Pull request by jamesadevine**

### CODE REVIEW

**Comment by jamesadevine** [comment]
Leaving open for now, likely more updates to come.

**Comment by pelikhan** [comment]
Do you still need join? https://github.com/microsoft/pxt-common-packages/compare/d629a1af5d45%5E...3de41e3bcdc2#diff-575174b47a0bceec357e84ec71d965b7R270

**Comment by pelikhan** [comment]
Is this ready?

**Comment by jamesadevine** [comment]
@pelikhan yes, we'll add the changes to jacdac we discussed last week another time.

**Comment by pelikhan** [comment]
Do i need these changes to start testing the debugger?

**Comment by jamesadevine** [comment]
To start testing the web usb debugger I'll need to bump codal-core for both SAMD51 / STM32

**Comment by jamesadevine** [comment]
@pelikhan Added long serial number to linux.

File: libs/core---linux/target.cpp [hunk 1]
```diff
@@ -124,13 +124,29 @@ int getSerialNumber() {
...
+    if (!bigSerialNumber)
+        bigSerialNumber = 0xf00d0042f00d0042;
+
+    return bigSerialNumber;
+}
+
+int getSerialNumber() {
+    static int serial;
+
+    if (serial)
```
Comment (mmoskal) [inline libs/core---linux/target.cpp]: can skip caching here

File: libs/game/console.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 namespace game.consoleOverlay {
     let consoleColor = 1;
     let consoleStrings: string[];
+    let tabSize = 8;
```
Comment (mmoskal) [inline libs/game/console.ts]: const?

File: libs/jacdac/buffer.ts [hunk 3]
```diff
@@ -267,5 +266,13 @@ namespace jacdac {
 
             Buffer.memmove(this.buffer, dstOffset, src.buffer, srcOffset, length)
         }
+
+        static join(b1: Buffer, b2: Buffer) : Buffer {
```
Comment (mmoskal) [inline libs/jacdac/buffer.ts]: tends to be called 'concat' in node.js etc

**Pull request merged**